src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_ARRAY_SIZE = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+    private int size;
+    private Object[] array;
+
+    ArrayList() {
+        array = new Object[INITIAL_ARRAY_SIZE];
+    }
+
+    private void grow(int value) {
+        if (size + value > MAX_ARRAY_SIZE) {
+            throw new OutOfMemoryError();
+        }
+        Object[] temp = new Object[size + value];
+        System.arraycopy(array, 0, temp, 0, size);
+        array = temp;
+    }
+
+    private void grow() {
+        grow(INITIAL_ARRAY_SIZE / 2);
```

Comment:
you resize list each time only on 5 element, but you should do it with 1.5 multipluer 
<img width="839" alt="image" src="https://user-images.githubusercontent.com/32339836/211200635-6c439533-5760-4d64-91a6-13c72f3e30f3.png">